### ISSUE TITLE
Syscalls are lost

### ISSUE THREAD

**Issue by christopherferreira19** [BinaryAnalysisPlatform/bap#959]
I am trying to write a binary analysis with bap but I encountered an issue: the syscalls does not appear in the bir.

Here is a minimal example to reproduce:
```C
void test_getpid() {
    register int syscall_no asm("rax") = 39;
    asm("syscall");
}
```
```Makefile
CFLAGS=-nostdlib

libtest.so: test.o
	gcc -o libtest.so test.o -shared
```
I ran `bap libtest.so -d' and the resulting output is:
```asm
00000036: program
00000029: sub deregister_tm_clones()
00000002: 
00000003: RDI := 0x4020
00000004: RAX := 0x4020
00000005: v13 := RAX - RDI
0000000b: ZF := 0 = v13
00000011: when ZF goto %0000000c
00000010: goto %0000000d

0000000c: 
0000000f: goto %00000023

0000000d: 
00000012: goto %00000013

00000013: 
00000014: RAX := mem[0x3FE0, el]:u64
00000015: v15 := RAX
0000001b: ZF := 0 = v15
00000021: when ZF goto %0000001c
00000020: goto %0000001d

0000001c: 
0000001f: goto %00000023

0000001d: 
00000022: goto %00000027

00000023: 
00000024: v19 := mem[RSP, el]:u64
00000025: RSP := RSP + 8
00000026: return v19

00000027: 
00000028: goto RAX


00000035: sub test_getpid()
0000002a: 
0000002b: v17 := RBP
0000002c: RSP := RSP - 8
0000002d: mem := mem with [RSP, el]:u64 <- v17
0000002e: RBP := RSP
0000002f: RAX := 0x27
00000030: RBP := mem[RSP, el]:u64
00000031: RSP := RSP + 8
00000032: v18 := mem[RSP, el]:u64
00000033: RSP := RSP + 8
00000034: return v18
```

I would expect the syscall instruction in between the two lines:
```asm
0000002f: RAX := 0x27
00000030: RBP := mem[RSP, el]:u64
```

**Comment by ivg** [comment]
yep, syscalls are for some reason lifted as `special` statements, which are erased later.

**Comment by ivg** [comment]
For the context. It is hard to reify a syscall into a correct interrupt, because the parameter itself is rather ABI specific. E.g., this is how the sycall is seen from the disassembler point of view, 
```
$ bap-mc '0f 05' --show-insn --show-bil 
SYSCALL()
{
  
}
```

Therefore, given the current BIL it doesn't really have any representable semantics. I can imagine a couple of workarounds, but I'm really reluctant to implement them because the crux of the problem is the old lifter framework (e.g., the lifting function is totally context independent, and have access only to the instruction and nothing more) together with BIL, which doesn't really support well resolving syscalls (like we have the `CpuExn` statement which is rather low-level and hard to lift correctly). 

So, I'm feeling like it would be a good idea to postpone the fix until the 2.0 release, in which the fix will be trivial.

**Comment by ivg** [comment]
#1059 made it is possible to translate syscalls into calls, e.g., 
```
$ bap-mc '0f 05' --show-insn --show-{bil,bir} --bil-enable-intrinsic=:special
SYSCALL()
{
  insn:next_address := 2
  special (bil-call:llvm-x86_64:SYSCALL)
}
00000007: 
00000008: insn:next_address := 2
0000000c: call %00000004 with noreturn
```

(Note, during the CFG reconstruction this `call %00000004` will be linked to `@llvm-x86_64:SYSCALL`).

This also works for other instructions that have special semantics in BIL, as well as for instructions that still lack semantics. It is possible to turn other instructions into calls if necessary, see #1059 for more information.
